File: LoanApplicationWebAPI/Models/Repayment.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace LoanApplicationWebAPI.Models
{
    public class Repayment
    {
        public int Id { get; set; }

        public int ApprovedLoanId { get; set; }
        public LoanApproved? ApprovedLoan { get; set; }

        public int InstallmentNumber { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal Amount { get; set; }

        public DateTime DueDate { get; set; }
        public bool IsPaid { get; set; }
        public DateTime? PaidDate { get; set; }
    }
}

File: src/app/Models/repayment.ts
import { LoanApproved } from "./loan-approved";

export interface Repayment {
  repaymentId?: number;
  installmentNumber: number;
  amount: number;
  dueDate: string;
  status: 'Pending' | 'Paid';
  loanApproved:LoanApproved;
}
